cobra/core/Model.py

```diff
@@ -118,6 +117,45 @@ def description(self, value):
         self.name = value
         warn("description deprecated", DeprecationWarning)
 
+    @property
+    def medium(self):
+        return {rxn.id: rxn.bounds for rxn in
+                self.reactions.query(lambda x: x, 'boundary')}
+
+    @medium.setter
+    def medium(self, medium):
+        """Get or set the constraints on the model exchanges.
+
+        `model.medium` returns a dictionary of the bounds for each of the
+        boundary reactions, in the form of `{rxn_id: (l_bound, u_bound)}`
+
+        Parameters
+        ----------
+        medium: dictionary-like
+            The medium to initialize. medium should be a dictionary defining
+            `{rxn_id: bounds}` pairs.
+
+        """
+
+        # Check the symmetric difference to see if reactions are missing
+        boundary_rxn_keys = set([rxn.id for rxn in self.reactions.query(
+                        lambda x: x.boundary)])
+        medium_keys = set(medium.keys())
+        missing_in_medium = boundary_rxn_keys - medium_keys
+        missing_in_model = medium_keys - boundary_rxn_keys
+
+        for rxnid in missing_in_medium:
+            warn("{} is missing from the medium".format(rxnid))
```

Comment:
So I think of it like this - when calling e.g. `model.set_medium()`, boundary reactions that are mentioned in that dict should be updated to allow import. If a flag is set boundary reactions that are not mentioned should be updated to be blocked as they are not available in the medium. I feel this would be consistent with thinking of medium as 'what you let your cells grow in' whereby not being able to set CO2 production perhaps isn't so intuitive use anyway. Warnings are then not necessary as the behaviour is explicit. Reactions not mentioned at all in the model but in the medium should raise a warning to cover the spell-error issue.

My first use-case in mind here is switching the C-source in my medium from glucose to galactose, in this case glucose import will have to be blocked and galactose enabled.

Would it be useful with a live discussion? I could organise another meeting.